narwhal/primary/src/metrics.rs

```diff
@@ -404,6 +406,13 @@ impl PrimaryMetrics {
                 registry
             )
             .unwrap(),
```

Comment:
Can we also keep track of the size of `pending_certificates`, and the total certificates received via request vote?

Reply:
Sure but maybe in separate PR? I think the total certificates received via RequestVote should already be computable right now as the total number of certificates in `tx_certificates` minus the number of SendMessage RPCs?

Reply:
Next PR sounds good. The rate of insertion into `tx_certificates` may be approximated with rate of processing from the channel.